internal/service/elasticsearch/domain_test.go

```diff
@@ -1164,6 +1175,27 @@ func TestAccElasticsearchDomain_Update_version(t *testing.T) {
 		}})
 }
 
+func TestAccElasticsearchDomain_disappears(t *testing.T) {
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	resourceName := "aws_elasticsearch_domain.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acctest.PreCheck(t); testAccPreCheckIamServiceLinkedRoleEs(t) },
+		ErrorCheck:   acctest.ErrorCheck(t, elasticsearch.EndpointsID),
+		Providers:    acctest.Providers,
+		CheckDestroy: testAccCheckDomainDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDomainConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					acctest.CheckResourceDisappears(acctest.Provider, tfelasticsearch.ResourceDomain(), resourceName),
+					acctest.CheckResourceDisappears(acctest.Provider, tfelasticsearch.ResourceDomain(), resourceName)),
```

Comment:
Small change needed:  this like looks to be a duplicate

Reply:
duplicate is on purpose to check the delete function is actually idempotent. (usually missing catching an error on delete) should probably be added into `CheckResourceDisappears` but could potentially break existing tests.